### DISCUSSION TITLE
how to pass variables to a loaded movieclip?

### DISCUSSION THREAD

**Discussion by ousia** [ruffle-rs/ruffle#17350] [Q&A]
Hi there,

I have the following two animations (please, remove the `.txt` extension [they are SWF files]):

1. A loader ([`pr-loader.swf.txt`](https://github.com/user-attachments/files/16479334/pr-loader.swf.txt).
1. A presentation ([`pr-presentation.swf.txt`](https://github.com/user-attachments/files/16479335/pr-presentation.swf.txt).

I may use flashvars with the presentation, such as in:

```
ruffle -P slide=3 pr-presentation.swf
```

This works fine. Although slides in this particular presentation are zero-based (and values for the `slide` variable are not).

I would like to invoke the same flashvar with the loader (`ruffle -P slide=3 pr-loader.swf`) to have it passed to the presentation.

The (stripped down) AS2 code to load the presentation in the loader reads:

``` js
if (slide != null && slide > 1) {
    the_presentation = "./pr-presentation.swf?slide=slide";
} else {
    the_presentation = "./pr-presentation.swf";
}

first_loading = new Object();

first_loading.onMouseDown = function () {
  load_swf = new MovieClipLoader();

  load_swf.loadClip(the_presentation, placeholder);

  listen_loading = new Object();
};

Mouse.addListener(first_loading);
```

I’m afraid I don’t know how to pass the value from the loader to the loaded movieclip (since this doesn’t work at all).

Would any kind soul know how to do that in AS2?

Many thanks for your help.

**Comment by colin-i** [comment]
Maybe something like this?
```
createEmptyMovieClip('load_swf',getNextHighestDepth());
load_swf.slide=slide;
load_swf.loadMovie('./pr-presentation.swf');
```

**Reply by ousia** [reply]
I’m afraid I’m not being able to work.

Many thanks for your help, @colin-i.
